### ISSUE TITLE
Why do `llvm-addr2line-16` and `llvm-addr2line-18` try connecting to canonical.com (185.125.188.100)?

### ISSUE THREAD

**Issue by gdh1995** [llvm/llvm-project#105606]
Hello I'm a newbie to LLVM and DWARF. In the past days I found that:
- `llvm-addr2line-16` pauses tens of seconds before outputting results for a small hello-world program
- `llvm-addr2line-18` pauses about 1 second before outputting
- while `llvm-addr2line-10` always finishes without any wait

Then today I did some tests and noticed there're `connect` and `poll` syscalls to an outside IP. When tried `strace llvm-addr2line-18 -afC -e a.out <<<$addr 2>&1 | grep 185.125`, it logged:
``` log
connect(5, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("185.125.188.100")}, 16) = -1 EINPROGRESS (Operation now in progress)
getpeername(5, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("185.125.188.100")}, [128 => 16]) = 0
getpeername(5, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("185.125.188.100")}, [128 => 16]) = 0
getpeername(5, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("185.125.188.100")}, [128 => 16]) = 0
```

And https://ipinfo.io/AS41231/185.125.188.0/23 said the IP address belongs to `canonical.com`.

The difference of `llvm-addr2line-16` is that it always trys to `poll` and `recvfrom` the socket, and when my internet connection is not so good it pauses for seconds. `llvm-addr2line-10` never connects to this IP.

Then I wonder:
- why does `llvm-addr2line` connect to an outside website (canonical.com:443) and wait for response?
- how may I disable this behavior?
  - I ran `grep canonical.com /etc /lib/x86_64-linux-gnu` but nothing useful was found.

My environment is `Ubuntu 24.04 LTS` and `LLVM 1:10.0.0-4ubuntu1, 1:16.0.6-23ubuntu4, 1:18.1.3-1ubuntu1`.

**Comment by philipc** [comment]
Do you have DEBUGINFOD_URLS set in your environment? debuginfod.ubuntu.com is at 185.125.188.100.

**Comment by gdh1995** [comment]
Um yes there is the `DEBUGINFOD_URLS=https://debuginfod.ubuntu.com`. And when I set it empty, `llvm-addr2line-18` gets much faster. Thank you very much!

I found `/etc/debuginfod/elfutils.urls` specifies the website, and `/usr/share/libdebuginfod-common/debuginfod.sh` reads it into the environment variable.

I set it empty in my `~/.zshrc` and then it's empty as expected. Then this issue is solved perfectly!
